### ISSUE TITLE
pdb CLI argument parsing errors with arguments to module targets

### ISSUE THREAD

**Issue by vivster7** [python/cpython#113548]
# Bug report

### Bug description:

The `pdb` CLI is no longer correctly forwarding arguments to other tools. 

Example reproduction:
- It raises an error `pdb: error: argument pyfile: not allowed with argument -m`
```
vscode ➜ ~ $ docker run -it --rm python:3.13.0a2 python -m pdb -m calendar 1
usage: pdb [-h] [-c command] (-m module | pyfile) [args ...]
pdb: error: argument pyfile: not allowed with argument -m
```

Expected behavior:
- It starts the `pdb` debugger.
```
vscode ➜ ~ $ docker run -it --rm python:3.12 python -m pdb -m calendar 1
> /usr/local/lib/python3.12/calendar.py(1)<module>()
-> """Calendar printing functions
(Pdb) 
```

### CPython versions tested on:

3.13

### Operating systems tested on:

Linux

### Linked PRs
* gh-113557

**Comment by vivster7** [comment]
https://github.com/python/cpython/pull/109165/files introduced this regression.

I think the core of the issue is we are trying to parse arguments that we just need to be forwarding. I think treating the `-m` argument like a boolean argument to indicate "module-mode"  and using `parse_known_args` would be closest to mimicking the behavior of  the previous `getopt` behavior.

**Comment by gaogaotiantian** [comment]
This is actually a symptom for the real serious issue - pdb can't deal with arguments to modules (-m) now.

```
# This will fail
./python -m pdb -m calendar 1
```

The problem is, with the current argument parsing method, `1` will be considered as the file to debug and it will conflict with `-m`. Making `-m` the boolean introduces other issues:

1. The help message would be confusing
2. Commands like `./python -m pdb calendar -m` will execute successfully, which is confusing.

As for the `-c` part, that's not an issue, because when you have ambiguous options, you have to use `--` to separate them clearly, that's true for any CLI and is not a pdb specific issue.

But yes, we need to figure out a way to deal with the arguments to `-m` and it just seems like we can't have a perfect solution with `argparse`. We need some tradeoff here.

**Comment by gaogaotiantian** [comment]
Fixed in #113557
